src/client/components/forms/userCollection.js

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (C) 2020 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import * as bootstrap from 'react-bootstrap';
+import CustomInput from '../../input';
+import EntitySearchFieldOption from '../../entity-editor/common/entity-search-field-option';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../input/select-wrapper';
+import classNames from 'classnames';
+import request from 'superagent';
+import {uniqBy} from 'lodash';
+
+
+const {Alert, Button, Col} = bootstrap;
+
+class UserCollectionForm extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const {collaborators, ...collection} = props.collection;
+		this.state = {
+			collaborators,
+			collection,
+			errorText: null
+		};
+
+		// React does not autobind non-React class methods
+		this.getCleanedCollaborators = this.getCleanedCollaborators.bind(this);
+		this.handleAddCollaborator = this.handleAddCollaborator.bind(this);
+		this.handleRemoveCollaborator = this.handleRemoveCollaborator.bind(this);
+		this.handleChangeCollaborator = this.handleChangeCollaborator.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	handleSubmit(evt) {
+		evt.preventDefault();
+
+		if (!this.isValid()) {
+			this.setState({
+				errorText: 'Incomplete Form'
+			});
+			return;
+		}
+
+		const collaborators = this.getCleanedCollaborators();
+		const description = this.description.getValue();
+		const name = this.name.getValue();
+		const privacy = this.privacy.getValue();
+		const entityType = this.entityType.getValue();
+
+		const data = {
+			collaborators,
+			description,
+			entityType,
+			name,
+			privacy
+		};
+
+		let submissionURL;
+		if (this.props.collection.id) {
+			submissionURL = `/collection/${this.props.collection.id}/edit/handler`;
+		}
+		else {
+			submissionURL = '/collection/create/handler';
+		}
+		request.post(submissionURL)
+			.send(data)
+			.then((res) => {
+				window.location.href = `/collection/${res.body.id}`;
+			}, (error) => {
+				this.setState({
+					errorText: 'Internal Error'
```

Comment:
I don't think we should show the internal error to the user. ( it won't make any sense to them anyway)
we can do print the error in the logs. Will be useful while debugging